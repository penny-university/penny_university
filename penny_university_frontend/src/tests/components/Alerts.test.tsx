import React from 'react'
import { shallow, mount } from 'enzyme'
import { Alert } from 'reactstrap'
import { ErrorAlert } from '../../components/alerts/index.tsx'

describe('error alert component', () => {
  it('renders correctly', () => {
    const error = { error: 'error' }
    const component = shallow(<ErrorAlert error={error} dismiss={jest.fn()} />)
    expect(component).toMatchSnapshot()
  })

  it('should display with error message', () => {
    const error = { error: 'error' }
    const component = mount(<ErrorAlert error={error} dismiss={jest.fn()} />)
    expect(component.find(Alert).text()).toContain('error')
  })

  it('should not display without an error message', () => {
    const component = shallow(<ErrorAlert error={undefined} dismiss={jest.fn()} />)
    expect(component.find(Alert)).toEqual({})
  })

  it('should be dismissed', () => {
    const error = { error: 'error' }
    const component = mount(<ErrorAlert error={error} dismiss={jest.fn()} />)
    component
      .find('button.close')
      .at(0)
      .simulate('click')

    expect(component.find(Alert)).toEqual({})
  })
})
