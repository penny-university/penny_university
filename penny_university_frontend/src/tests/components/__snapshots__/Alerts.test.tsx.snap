// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error alert component renders correctly 1`] = `
<div
  className="alert-container"
>
  <Alert
    closeAriaLabel="Close"
    color="danger"
    fade={true}
    isOpen={true}
    tag="div"
    toggle={[Function]}
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    <p
      className="mb-0"
      key="ErrorMessage-0"
    >
      error
    </p>
  </Alert>
</div>
`;
